### PR TITLE
Fix bottom overflow error in about tab

### LINKED ISSUE #49: Bottom Overflowed by 97 pixels
**Describe the bug**
The bottom is overflowed by 97 pixels in About tab

**To Reproduce**

1. Go to Events or Reps tab
2. Click on the search bar to open the keyboard
3. Go to About tab 

**Expected behavior**

The keyboard shouldnot be opened when user is switching to About tab

**Screenshots**
Screenshot_1571234022

**Additional context**

To solve the issue, one can remove the focus when the index of the tab controller is changed.

### PR DESCRIPTION
**Pull request by ankank30**
Fixes #49  

**Summary**: 

SingleChildScrollView() was used in the about page since there is a possibility that screen size is too small to accommodate all elements (here Text).

**Screenshots for the change**: 

Before: 
Screenshot_1571255424

After: 
Screenshot_1571254757

### CODE REVIEW

File: reps_event_app/lib/ui/about.dart [hunk 1]
```diff
@@ -15,20 +15,24 @@ class _AboutState extends State<About> {
...
-          Padding(
-            padding: const EdgeInsets.only(top:20.0, bottom: 16.0, left: 16.0, right: 16.0),
-            child: Text(
-              Utils.about_info,
-              style: TextStyle(
-                fontSize: 28,
-                fontFamily: Utils.zilla_slab,
+      //SingleChildScrollView is used to handle cases when screen size is small
+      //enough to have all elements (Text) or fit completely.
+      body: SingleChildScrollView(
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/ui/about.dart]: `SingleChildScrollView` is not making the screen responsive, it's making the screen scrollable. The user should not be able to Scroll in the About tab.
Screenshot_1571290335

The main issue that needs to be addressed is to hide the keyboard in the About tab. You can hide the keyboard by removing the focus when tabs are changed.

**Review by AyushBherwani1998 — CHANGES_REQUESTED** [review]

**Comment by ankank30** [comment]
I have added code to remove the keyboard in 2 places. `onTap` in `TabBar` widget starts to remove the keyboard before switching the tab and if the about tab is reached using swipe then `build` function removes the keyboard.

Also now the text in about page is responsive to the screen size.

**Comment by ankank30** [comment]
untitled

File: reps_event_app/lib/ui/about.dart [hunk 2]
```diff
@@ -10,29 +10,32 @@ class About extends StatefulWidget {
 class _AboutState extends State<About> {
   String dateTime;
   double appBarSize;
+  double kFontSize;
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/ui/about.dart]: can you please describe why is the variable named `kFontSize`

File: reps_event_app/lib/ui/events.dart [hunk 3]
```diff
@@ -60,6 +60,11 @@ class _EventsState extends State<Events> {
           ],
           bottom: TabBar(
             indicatorColor: Colors.white,
+            onTap: (kTabLabel) {
```
Comment (AyushBherwani1998) [inline reps_event_app/lib/ui/events.dart:63]: The keyboard will hide when the user is switching the tab to About Section.  For a better UX can we hide the keyboard when the user is switching the tab from 0 to 1 or vice versa?
To do so, just remove the `KTabLabel == 2` and it should work fine.

Comment (ankank30) [reply reps_event_app/lib/ui/events.dart:63]: But still, if the user swipes to go to Reps tab then keyboard won't hide.

If you really want that whenever a tab is changed the keyboard should hide then I can get focus check in the starting of `build` function.

Comment (AyushBherwani1998) [reply reps_event_app/lib/ui/events.dart:63]: Yes, the keyboard won't hide during the swipe. 
But, adding the check mentioned below in the `build` function won't allow users to open the keyboard. You may give it a try.
```
if (FocusScope.of(context).hasPrimaryFocus) {
      FocusScope.of(context).unfocus();
   }
```

Comment (ankank30) [reply reps_event_app/lib/ui/events.dart:63]: I tried but as you said it wasn't allowing the keyboard to open.

Comment (AyushBherwani1998) [reply reps_event_app/lib/ui/events.dart:63]: > 
> 
> I tried but as you said it wasn't allowing the keyboard to open.

Try this, create a `TabController _tabController` and in the `initState` method attach a listener to `_tabController` to listen to tab changes and in the listener function remove the focus.

Comment (ankank30) [reply reps_event_app/lib/ui/events.dart:63]: Ya, I even tried that but again it was working for the taps but not for swipe. Moreover, it was making the tab changes unsmooth.

**Review by AyushBherwani1998 — APPROVED** [review]

File: reps_event_app/lib/ui/events.dart (hunk 3 above)
Comment (AyushBherwani1998) [reply reps_event_app/lib/ui/events.dart:63]: Okay, I'll merge this request and open a new issue for swipe.
Thank you for your contribution.

**Review by AyushBherwani1998 — APPROVED** [review]

**Pull request merged**